### ISSUE TITLE
clang-build fails with "Illegal character in Path"

### ISSUE THREAD

**Issue by FlorianReimold** [Caphyon/clang-power-tools#1248]
When I clang-build a VS Project, it fails for me with the following message (German Windows):

```
--- CLANG TIDY STARTED ---

Get-ChildItem : Illegales Zeichen im Pfad.
In C:\users\uid38621\appdata\local\microsoft\visualstudio\16.0_cec81720\extensions\z2xrjrot.1qu\Tooling\v1\clang-build.
ps1:545 Zeichen:12
+    $slns = Get-ChildItem -recurse -LiteralPath $pathToCheck -Filter " ...
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-ChildItem], ArgumentException
    + FullyQualifiedErrorId : System.ArgumentException,Microsoft.PowerShell.Commands.GetChildItemCommand
 
Cannot find given project(s)

--- CLANG TIDY FINISHED ---
```

I tracked down the issue to the following line of code:

https://github.com/Caphyon/clang-power-tools/blob/041844f6f3e1f3e11b4c5736eba8c38975cfaa49/ClangPowerTools/ClangPowerTools/Tooling/v1/clang-build.ps1#L545

I believe, this is due to adding `\\?\` and running the command in PowerShell x86. While I can run the command in x64 Powershell, it fails in x86:

![image](https://user-images.githubusercontent.com/11774314/216253682-52c65bb3-d01e-4a50-8d40-42045fd85067.png)
![image](https://user-images.githubusercontent.com/11774314/216253700-b8eb9bba-86e8-455f-8827-f3825b0e5585.png)

I literally just removed the code adding the `\\?\` prefix for making it work:

https://github.com/Caphyon/clang-power-tools/blob/041844f6f3e1f3e11b4c5736eba8c38975cfaa49/ClangPowerTools/ClangPowerTools/Tooling/v1/clang-build.ps1#L538-L543

I didn't draft a PR, as there is probably a reason for this code to exist. However it looks to me like it either needs to be removed, or the Addin must make sure it runs in x64 Powershell.

----

- Win 10 x64
- VS 2019
- Project configured with CMake
- Powershell 5
    ```
    Name                           Value
    ----                           -----
    PSVersion                      5.1.19041.2364
    PSEdition                      Desktop
    PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
    BuildVersion                   10.0.19041.2364
    CLRVersion                     4.0.30319.42000
    WSManStackVersion              3.0
    PSRemotingProtocolVersion      2.3
    SerializationVersion           1.1.0.1
    ```

**Comment by mariru27** [comment]
Hi @FlorianReimold,

Thank you for reporting this issue, and for provided information
We are looking into it and working towards a fix. We will keep you updated as progress is made.

Best regards,
Marina

**Comment by FlorianReimold** [comment]
Hi @mariru27 

I tried it out, but it didn't work:
![image](https://user-images.githubusercontent.com/11774314/216282908-06e06fe9-b10a-4d52-b2be-be7e2fc525b7.png)

**Comment by mariru27** [comment]
Thank you @FlorianReimold, now I am sure that the problem is from script

We will make an fix based on your provided information 

Kind regards,
Marina

**Comment by mariru27** [comment]
Hi @FlorianReimold,

We made a fix, but after more attempts we can't reproduce your problem, that means that we can't integrate the fix
Can you run clang-tidy on a small test project with verbose mode on, and send us the log?

Best regards,
Marina

**Comment by FlorianReimold** [comment]
Sure! Any new version I should try? Or shall I try with the Version from your google drive link?

Oh, and did you try out the command in x86 vs x64 Powershell? I am wondering if it is something on my PC only, because that different behavior is kind of strange.

**Comment by mariru27** [comment]
Yes, we tried the command in x86 vs x64 Powershell, but it works for us

You can use the version from Google Drive

**Comment by FlorianReimold** [comment]
Oh man, this is cursed...

So I tried the original script and it works now. I have no Idea why. 

But I tried the command again, in both x64 Powershell and x86 Powershell. Today it fails in x64 Powershell, but only in 1 Window and I cannot reproduce it with the other ones. But in that window the command keeps failing, even if I paste it again, just as in in the other windows.

![grafik](https://user-images.githubusercontent.com/11774314/216591958-24f66eff-6171-439f-840a-8ce2dc662997.png)

The upper and lower window are x86 powershells, the middle ones are x64 ones. 

Shall I still create a verbose log for you?
